pkg/serviceaccount/legacy.go

```diff
@@ -144,12 +145,28 @@ func (v *legacyValidator) Validate(ctx context.Context, tokenData string, public
 			return nil, fmt.Errorf("ServiceAccount UID (%s) does not match claim (%s)", serviceAccount.UID, serviceAccountUID)
 		}
 
+		// Track secret-based long-lived service account tokens and add audit annotations and metrics.
+		autoGenerated := false
+
+		// Check if it is an auto-generated secret-based token
 		for _, ref := range serviceAccount.Secrets {
 			if ref.Name == secret.Name {
+				autoGenerated = true
 				warning.AddWarning(ctx, "", "Use tokens from the TokenRequest API or manually created secret-based tokens instead of auto-generated secret-based tokens.")
+				auditInfo := fmt.Sprintf("%s uses auto-generated secret-based long lived tokens: %s", serviceAccount.Name, secret.Name)
+				audit.AddAuditAnnotation(ctx, "authentication.k8s.io/auto-generated-token", auditInfo)
```

Comment:
we already have the service account info in the audit log (it's the authenticated user), just use the name of the secret as the audit data, no need for a long human-readable message

```suggestion
				audit.AddAuditAnnotation(ctx, "authentication.k8s.io/legacy-token-autogenerated-secret", secret.Name)
```